File: TrinhsGroup/View/Model/WooErrorResponse.swift
//
//  WooErrorResponse.swift
//  TrinhsGroup
//
//  Created by longnh on 2025/04/08.
//

import Foundation

struct WooErrorResponse: Codable, LocalizedError {
    let code: String
    let message: String
    let data: WooErrorData?

    var errorDescription: String? {
        "\(code): \(message)"
    }
}

struct WooErrorData: Codable {
    let status: Int?
}

File: kmp/shared/src/commonMain/kotlin/com/trinhsgroup/shared/model/WooErrorResponse.kt
package com.trinhsgroup.shared.model

import kotlinx.serialization.Serializable

/**
 * WooCommerce error response wrapper.
 * Mirrors Swift's WooErrorResponse struct.
 *
 * Extends Exception so it can be thrown directly.
 */
@Serializable
data class WooErrorResponse(
    val code: String,
    override val message: String,
    val data: WooErrorData? = null
) : Exception(message)

/**
 * Additional error data from WooCommerce API.
 * Mirrors Swift's WooErrorData struct.
 */
@Serializable
data class WooErrorData(
    val status: Int? = null
)
